tools/file_packager.py

```diff
@@ -218,6 +224,9 @@ def main():
     elif arg == '--no-node':
       support_node = False
       leading = ''
+    elif arg == '--wasmfs-c':
```

Comment:
Rather than adding two different modes I wonder if we can instead just move all embedding to C/C++ code, I don't see why it shouldn't work for the JS filesystem too.

Reply:
That way perhaps we could delete the old JS codegen from here?

Reply:
There is a trade off that the Wasm memory may be more valuable than JS heap memory and it also can't be garbage collected. I'm not sure whether people are preloading file large enough that it would matter, though.

Reply:
We are talking about embedding only here, and not preloading.   In that case, we are comparing embedded binaries in JS as base64 strings with wasm memory.  I doubt those strings ever get collected.. and they are a lot larger than the equivalent in wasm memory because they are base64. 

The other big advantage of using wasm memory is that it opens the door for zero-copy files (at least in read-only circumstances)... even for the old FS I think this is do-able.